src/internal/operations/backup.go

```diff
@@ -335,28 +339,28 @@ func collectMetadata(
 	return dcs, nil
 }
 
-func selectorToOwnersCats(sel selectors.Selector) *kopia.OwnersCats {
+func selectorToReasons(sel selectors.Selector) []kopia.Reason {
 	service := sel.PathService()
-	oc := &kopia.OwnersCats{
-		ResourceOwners: map[string]struct{}{},
-		ServiceCats:    map[string]kopia.ServiceCat{},
-	}
-
-	oc.ResourceOwners[sel.DiscreteOwner] = struct{}{}
+	reasons := []kopia.Reason{}
 
 	pcs, err := sel.PathCategories()
 	if err != nil {
-		return &kopia.OwnersCats{}
+		// This is technically safe, it's just that the resulting backup won't be
+		// usable as a base for future incremental backups.
+		return nil
 	}
 
 	for _, sl := range [][]path.CategoryType{pcs.Includes, pcs.Filters} {
 		for _, cat := range sl {
-			k, v := kopia.MakeServiceCat(service, cat)
-			oc.ServiceCats[k] = v
+			reasons = append(reasons, kopia.Reason{
+				ResourceOwner: sel.DiscreteOwner,
+				Service:       service,
+				Category:      cat,
+			})
```

Comment:
Thinking out loud: is there ever a time where the list of reasons will have different owners or services?  It feels like a []Reasons is more accurately represented as something like:
```go
type Reason struct {
  ResourceOwner string
  Service path.ServiceType
  Categories []path.CategoryType
}

Reply:
>is there ever a time where the list of reasons will have different owners or services?

Not at the moment and we don't seem to have anything like that planned

However, after going through iterations of code with `OwnersCats` (which was similar to what you've proposed) and `Reason`, I kind of like the simplicity of the `Reason` struct. There's no gotchas that need to be known about when iterating through them and it only requires a single for-loop instead of a double for-loop. It also better matches the unit of granularity we need to use for delta token-base incrementals: the (resource owner, service, category) tuple. We're restricted to checking this tuple because that's the granularity that the CLI and selectors allow when making backups

Reply:
> only requires a single for-loop instead of a double for-loop

I was imagining this as only passing around a single Reason, not multiples.  The only loop would be over the categories.

Per granularity along the tuple and existence of gotchas, I do think there's a possibility.  Within an operation, we organize data into owner+service silos.  There is no case where we intend to handle multiple users/services at a time.  Multiple reasons with independent declarations provide handling as if that were possible.

Reply:
>Multiple reasons with independent declarations provide handling as if that were possible.

I guess I don't see that a drawback really 😅 If we want to ensure it doesn't happen we can just add a validation check to ensure there's only one service/resource owner

> I was imagining this as only passing around a single Reason, not multiples.

that sounds pretty similar to `OwnersCats`. The `snapshot_manager` still needs to break it down into individual (resource owner, service, category) tuples because the CLI still allows that granularity. That means we may have different bases for something like `backup create exchange --data email,contacts` where each selected base will have some (possibly complete) subset of the categories that were sourced from the original

Reply:
> I guess I don't see that a drawback really
That's fair.  More of a talking point than a concern, really.  All other points are good.  Thanks for the chat.